### PR TITLE
Support "Uncategorized" extensions

### LINKED ISSUE #21350: Support "Uncategorized" extensions
### Description

Currently extensions without category are not processed and not visible in code.quarkus.io

### Implementation ideas

Use "uncategorized" as category here (if empty): https://github.com/quarkusio/quarkus/blob/main/independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java#L25

Add "Uncategorized" as category last in `orderedCategories` here: https://github.com/quarkusio/quarkus/blob/main/independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java#L33

### PR DESCRIPTION
**Pull request by geekiyer**
Fixes #21350

### CODE REVIEW

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java [hunk 1]
```diff
@@ -35,6 +42,9 @@ public static CatalogProcessor of(ExtensionCatalog catalog) {
                 orderedCategories.add(new ProcessedCategory(c, extsByCategory.get(c.getId())));
             }
         }
+        JsonCategory category = new JsonCategory();
+        category.setId(UNCATEGORIZED);
```
Comment (ia3andy) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java:46]: Could you also set name: `Uncategorized` and description: `The category is not defined for those extensions.`

**Review by ia3andy — CHANGES_REQUESTED** [review]

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java (hunk 1 above)
Comment (geekiyer) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java]: Done!

**Review by ia3andy — CHANGES_REQUESTED** [review]
Could you squash to 1 commit please, then it's good to go :) Thanks!

**Comment by ia3andy** [comment]
@geekiyer  you also need to run `mvn process-sources` to clean the code and fix the build :)

**Comment by ia3andy** [comment]
@geekiyer could you update this PR, it is really needed.

**Comment by geekiyer** [comment]
> @geekiyer could you update this PR please, it is really needed.

On it, sorry!

**Comment by geekiyer** [comment]
> @geekiyer you also need to run `mvn process-sources` to clean the code and fix the build :)

Will do! Although, I'd suggest this should happen on CI automatically with every PR when needed.. 🙂

**Comment by geekiyer** [comment]
> Could you squash to 1 commit please, then it's good to go :) Thanks!

Will do! Suggestion to use Github's in-built, Squash and Merge feature? 🙂

**Comment by geekiyer** [comment]
Will open a new PR with a single commit! Facing some issues while squashing commits.

**Pull request closed without merging**